### ISSUE TITLE
DVCFileSystem.get() doesn't parallelize download

### ISSUE THREAD

**Issue by grahameth** [iterative/dvc#8896]
# Bug Report

Calling get on DVCFileSystem with multiple files doesn't parallelize the download.

### Reproduce

This example downloads multiple files with a single get call:
```
fs = DVCFileSystem(repo_url, rev=rev)
remote_files = [...]
local_files  = [...]
fs.get(remote_files, local_files, callback=cb)
```
It works, but it downloads 1 file at a time.

### Expected

This example does the same, but accesses internal datastructures to make it faster:
```
fs = DVCFileSystem(repo_url, rev=rev)
remote_files = [...]
local_files  = [...]
fs._datafss[()].get(remote_files, local_files, callback=cb)
```
This internal get function parallelizes the download, and is thus about twice as fast on my machine.

Given that this functionality is already implemented, I find it a bit sad that it isn't used.

```
DVC version: 2.38.1 (conda)
---------------------------------
Platform: Python 3.9.15 on Windows-10-10.0.19045-SP0
Subprojects:
        dvc_data = 0.28.4
        dvc_objects = 0.14.0
        dvc_render = 0.0.15
        dvc_task = 0.1.8
        dvclive = 1.3.0
        scmrepo = 0.1.4
Supports:
        http (aiohttp = 3.8.3, aiohttp-retry = 2.8.3),
        https (aiohttp = 3.8.3, aiohttp-retry = 2.8.3),
        ssh (sshfs = 0.0.0)
```

**Comment by skshetry** [comment]
I'll suggest something like follows as a temporary workaround:
```python
with Repo.open(repo_url, rev=rev) as repo:
	repo.dvcfs.get(...)
```

There are some inconsistencies between `repo.dvcfs` and `DvcFileSystem`, but for the purpose of `fs.get()`, they should work similarly (and be faster). They are still a bit inefficient, ideally we want to take the underlying filesystem of the remote (say `s3fs`) directly for better performance.

**Comment by skshetry** [comment]
Fixed by #10540.
